### PR TITLE
New signature algorithm: ECC384

### PR DESCRIPTION
**Pull request by danielinux**
- [x] Add support in wolfBoot
- [x] Add support in C key tools
- [x] Add support in Python key tools
- [x] Add memory allocators in xmalloc.c for SMALL_STACK
- [x] Add memory allocators in xmalloc.c for SMALL_STACK+FAST_MATH
- [x] Add 'factory' test
- [x] Add 'update' test
- [x] Add 'corrupted' test
- [x] Add 'bad version' test

### CODE REVIEW

File: tools/keytools/sign.c [hunk 1]
```diff
@@ -353,9 +358,30 @@ static uint8_t *load_key(uint8_t **key_buffer, uint32_t *key_buffer_sz,
                     *pubkey_sz = 64;
                 }
             }
+        } else if (CMD.sign == SIGN_ECC384) {
+            ret = wc_ecc_init(&key.ecc);
+            if (ret == 0) {
+                ret = wc_ecc_import_unsigned(&key.ecc, *key_buffer, (*key_buffer) + 48,
+                        (*key_buffer) + 96, ECC_SECP384R1);
+                if (ret == 0) {
+                    *pubkey = *key_buffer; /* first 64 bytes is public portion */
```
Comment (dgarske) [inline tools/keytools/sign.c:367]: Please update comment. First 96 bytes.

File: tools/keytools/sign.c [hunk 2]
```diff
@@ -353,9 +358,30 @@ static uint8_t *load_key(uint8_t **key_buffer, uint32_t *key_buffer_sz,
                     *pubkey_sz = 64;
                 }
             }
+        } else if (CMD.sign == SIGN_ECC384) {
+            ret = wc_ecc_init(&key.ecc);
+            if (ret == 0) {
+                ret = wc_ecc_import_unsigned(&key.ecc, *key_buffer, (*key_buffer) + 48,
```
Comment (dgarske) [inline tools/keytools/sign.c:364]: Line length check

File: tools/keytools/sign.c [hunk 3]
```diff
@@ -353,9 +358,30 @@ static uint8_t *load_key(uint8_t **key_buffer, uint32_t *key_buffer_sz,
...
+                        (*key_buffer) + 96, ECC_SECP384R1);
+                if (ret == 0) {
+                    *pubkey = *key_buffer; /* first 64 bytes is public portion */
+                    *pubkey_sz = 96;
+                }
+            }
+        } else if (CMD.sign == SIGN_ECC521) {
+            ret = wc_ecc_init(&key.ecc);
+            if (ret == 0) {
+                ret = wc_ecc_import_unsigned(&key.ecc, *key_buffer, (*key_buffer) + 66,
```
Comment (dgarske) [inline tools/keytools/sign.c:374]: Line length check

File: tools/keytools/sign.c [hunk 4]
```diff
@@ -353,9 +358,30 @@ static uint8_t *load_key(uint8_t **key_buffer, uint32_t *key_buffer_sz,
...
+                        (*key_buffer) + 132, ECC_SECP521R1);
+                if (ret == 0) {
+                    *pubkey = *key_buffer; /* first 64 bytes is public portion */
```
Comment (dgarske) [inline tools/keytools/sign.c:377]: Please update comment. not 64 bytes.

**Review by dgarske — CHANGES_REQUESTED** [review]
Great work! Just a few minor cleanups. Also do you want to squash this PR?

**Comment by danielinux** [comment]
Good point, I'll squash these commits after your final review. I left them separate for now, for a more comfortable check of the changes suggested for `sign.c` in 886ad0f

**Review by dgarske — APPROVED** [review]
Thanks

**Pull request merged**